auditbeat/tracing/perfevent.go

```diff
@@ -492,7 +520,12 @@ func (m *recordMerger) readSampleNonBlock(ev *perf.Event, ctx context.Context) (
 	return nil, true
 }
 
-func pollAll(evs []*perf.Event, timeout time.Duration) (active int, closed int, err error) {
+func pollAll(evs []*perf.Event, timeout time.Duration) (int, int, error) {
```

Comment:
Suggest reverting (to `func pollAll(evs []*perf.Event, timeout time.Duration) (active, closed int, err error) {`) this change since the labels for the return params list are documentational. Retain the named returns below.

Reply:
although I am not a huge fan of named returns, improving readability through documentation is a higher prio than my code preferences, so reverting it is. However, the linter is going to complain about a naked return which is a false positive, are we okay with that? :)

Reply:
The changes below should remain, so this would keep the named returns.

Reply:
To be explicit...
```
func pollAll(evs []*perf.Event, timeout time.Duration) (active, closed int, err error) {
	pollfds := make([]unix.PollFd, len(evs))
	for idx, ev := range evs {
		fd, err := ev.FD()
		if err != nil {
			return 0, 0, errors.New("failed to get descriptor for perf event channel")
		}
		pollfds[idx] = unix.PollFd{Fd: int32(fd), Events: unix.POLLIN}
	}
	ts := unix.NsecToTimespec(timeout.Nanoseconds())

	for err = unix.EINTR; errors.Is(err, unix.EINTR); {
		_, err = unix.Ppoll(pollfds, &ts, nil)
	}
	if err != nil {
		return 0, 0, os.NewSyscallError("poll", err)
	}

	for _, fd := range pollfds {
		if fd.Revents&unix.POLLIN != 0 {
			active++
		}
		if fd.Revents&unix.POLLHUP != 0 {
			closed++
		}
	}
	return active, closed, err
}
```

Reply:
got yah!! so return explicitly the named returns